apps/fdsnws/share/js/event-time-selection.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Event-based time selection functionality for FDSNWS DataSelect builder
+ */
+
+// Simple debounce utility
+function debounce(fn, delay) {
+  let timeout;
+  return function(...args) {
+    clearTimeout(timeout);
+    timeout = setTimeout(() => fn.apply(this, args), delay);
+  };
+}
+
+// Event service availability
+let eventServiceAvailable = null; // null = unknown, true/false = known
+
+async function checkEventService() {
+  if (eventServiceAvailable !== null) return eventServiceAvailable;
```

Comment:
`checkEventService` is now only being called once so you can get rid of this `eventServiceAvailable` cache.